### PR TITLE
Ensure that concurrently creating collaborations with the same name r…

### PR DESCRIPTION
**Pull request by dirkmc**
…eturns the same collaboration

### CODE REVIEW

**Comment by dirkmc** [comment]
I was having an issue where if I called the `app.collaborate()` method twice, it wouldn't wait for the collaboration to start on the second method call, and so `collaboration.shared` was undefined

File: test/app.spec.js [hunk 1]
```diff
@@ -46,5 +46,17 @@ describe('app', function () {
     expect(await app.collaborate('collaboration name', 'fake', collaborationOptions)).to.equal(collaboration)
   })
 
+  it('can get concurrently created collaboration shared', async () => {
```
Comment (pgte) [inline test/app.spec.js]: I added this test to my current master (without your changes to `src/app.js`) and the test passes fine.

**Comment by dirkmc** [comment]
The problem is that it adds the newly created collaboration to `this._collaborations` (line 84) and then does `await collaboration.start()`. So if another thread calls `app.collaborate()` before the `await` has completed, the other thread will find the new collaboration in `this._collaborations` (line 73) and immediately return even though the collaboration hasn't started yet.

I found this issue when running my code with a remote rendezvous server, so it may be hard to repro locally, I'll try to write a test that fails consistently with the old code.

**Comment by pgte** [comment]
@dirkmc oh, the problem might be that you return an unstarted collaboration.
In this case, I think we can simply always `await collaboration.start()` before returning it.

A failing test case would be nice though :)

**Comment by pgte** [comment]
@dirkmc also, if you want to run only one test, you can:

1) Start a rendez-vous server locally (invoking rendezvous executable in `node_modules/.bin/rendezvous`).
2) mocha <test file>

**Comment by dirkmc** [comment]
Yes you're right, no need for such a complex solution. I included a test case that fails consistently with the old code :)

**Comment by dirkmc** [comment]
@pgte anything else you need for this PR?

**Comment by dirkmc** [comment]
Thanks

**Comment by pgte** [comment]
@dirkmc Thank you! Landed on v0.7.1. 🚢

**Pull request merged**